qtconsole/console_widget.py

```diff
@@ -43,6 +45,17 @@ def is_whitespace(char):
 #-----------------------------------------------------------------------------
 # Classes
 #-----------------------------------------------------------------------------
+class ShortcutManager(HasTraits):
+        # Define traits for shortcuts
+        shortcut_print = Unicode('Ctrl+P').tag(config=True)
+        shortcut_select_all = Unicode('Ctrl+A').tag(config=True)
+        shortcut_cut = Unicode(QtGui.QKeySequence(QtGui.QKeySequence.Cut).toString()).tag(config=True)
+        shortcut_copy = Unicode(QtGui.QKeySequence(QtGui.QKeySequence.Copy).toString()).tag(config=True)
+        shortcut_paste = Unicode(QtGui.QKeySequence(QtGui.QKeySequence.Paste).toString()).tag(config=True)
+
+        @observe('shortcut_print', 'shortcut_select_all','shortcut_cut','shortcut_copy','shortcut_paste')
+        def _on_shortcut_changed(self, change):
+            print(f"Shortcut for {change['name']} changed to: {change['new']}")
```

Comment:
So checking locally, I think the reason why adding this class causes a segfault is due to the usage of `QKeySequence` to define default values. The documentation of `QKeySequence` mentions that [if there is not a `QApplication` instance initialized calling `QKeySequence` can cause your application to crash](https://doc.qt.io/qt-6/qkeysequence.html#gnu-emacs-style-key-sequences). To prevent this I think we could use the traitlet `default` decorator. So this class definition will end up looking something like:

```suggestion

class ShortcutManager(HasTraits):
    """Default shortcuts definition and changes event handler."""

    # Define traits for shortcuts
    shortcut_print = Unicode('Ctrl+P').tag(config=True)
    shortcut_select_all = Unicode('Ctrl+A').tag(config=True)
    shortcut_cut = Unicode().tag(config=True)
    shortcut_copy = Unicode().tag(config=True)
    shortcut_paste = Unicode().tag(config=True)

    @default('shortcut_cut')
    def _default_shortcut_cut(self):
        return QtGui.QKeySequence(QtGui.QKeySequence.Cut).toString()

    @default('shortcut_copy')
    def _default_shortcut_copy(self):
        return QtGui.QKeySequence(QtGui.QKeySequence.Copy).toString()

    @default('shortcut_paste')
    def _default_shortcut_paste(self):
        return QtGui.QKeySequence(QtGui.QKeySequence.Paste).toString()

    @observe('shortcut_print', 'shortcut_select_all','shortcut_cut','shortcut_copy','shortcut_paste')
    def _on_shortcut_changed(self, change):
        print(f"Shortcut for {change['name']} changed to: {change['new']}")


```